content-based-message-routing/main.bal

```diff
@@ -0,0 +1,92 @@
+import ballerina/http;
+import ballerina/log;
+
+configurable int port = 8290;
+
+final http:Client grandOakEP = check initializeHttpClient("http://localhost:9090/grandoaks/categories");
+final http:Client clemencyEP = check initializeHttpClient("http://localhost:9090/clemency/categories");
+final http:Client pineValleyEP = check initializeHttpClient("http://localhost:9090/pinevalley/categories");
+
+function initializeHttpClient(string url) returns http:Client|error => new (url);
+
+type Patient record {|
+    string name;
+    string dob;
+    string ssn;
+    string address;
+    string phone;
+    string email;
+|};
+
+type ReservationRequest record {|
+    Patient patient;
+    string doctor;
+    string hospital_id;
+    string hospital;
+    string appointment_date;
+|};
+
+type Doctor record {|
+    string name;
+    string hospital;
+    string category;
+    string availability;
+    float fee;
+|};
+
+type ReservationResponse record {|
+    int appointmentNumber;
+    Doctor doctor;
+    Patient patient;
+    float fee;
+    string hospital;
+    boolean confirmed;
+    string appointmentDate;
+|};
+
+enum HospitalIds {
+    GRANDOAKS = "grandoaks",
+    CLEMENCY = "clemency",
+    PINEVALLEY = "pinevalley"
+};
+
+service /healthcare on new http:Listener(port) {
+    resource function post categories/[string category]/reserve(ReservationRequest payload)
+            returns ReservationResponse|http:NotFound|http:BadRequest|http:InternalServerError {
+        ReservationRequest {hospital_id, patient, ...reservationRequest} = payload;
+        http:Client hospitalEP;
+        match hospital_id {
+            GRANDOAKS => {
+                log:printInfo("Routed to Grand Oak Community Hospital");
```

Comment:
Shall we say routing to instead?

Also can be debug logs IMO.

Reply:
Include some info that will help uniquely identify the request.

Reply:
Think of it in this way. The logs help troubleshoot issues, and in the current form they won't be of much help.

The only exception is the first tutorial since it is a basic one and we want to show the flow.

Reply:
Good point. I thought of having a debug level logging before coming to the match statement and remove the logging while assigning hospital services.
```
        log:printDebug("Routing to a hospital", hospital_id = hospital_id);
        http:Client hospitalEP;
        match hospital_id {
            GRAND_OAK => {
                hospitalEP = grandOakEP;
            }
            CLEMENCY => {
                hospitalEP = clemencyEP;
            }
            _ => {
                hospitalEP = pineValleyEP;
            }
        }
```
WDYT?

Reply:
We can have it even within the clauses IMO. I'm OK either way. Need to include more info though. E.g., patient and doctor name.